Add two troubleshooters (#1249)
dotnet/docs-aspire: docs/troubleshooting/connection-string-missing.md

---
title: The connection string is missing
description: Learn how to troubleshoot the error "ConnectionString is missing" during execution of your app.
ms.date: 07/03/2024
---

# The specified name is already in use

> 

Reviewer: We strive for an introductory paragraph, sometimes a full paragraph isn't necessary, but we need at least a single sentence between headings.

## Symptoms

When your app accesses a service that needs one of the components in your app, it may fail with an exception similar to the following:

> "InvalidOperationException: ConnectionString is missing."

## Possible solutions

> Verify that the name of the resource, for instance a database resource, is the same in the AppHost and the Service that fails. 

Reviewer: ```suggestion
Verify that the name of the resource, for instance a database resource, is the same in the AppHost and the Service that fails.
```

> For example, if the AppHost defines a PostgreSQL resource with the name db1 like this: 

Reviewer: ```suggestion
For example, if the AppHost defines a PostgreSQL resource with the name `db1` like this: 
```

---
title: The connection string is missing
description: Learn how to troubleshoot the error "ConnectionString is missing" during execution of your app.
ms.date: 07/03/2024
---

# The specified name is already in use

> In an Aspire app, code identifies resources with an arbitrary string, such as "database". Code that is consuming the resource elsewhere must use the same string or it will fail.

Reviewer: ```suggestion
In .NET Aspire, code identifies resources with an arbitrary string, such as "database". Code that is consuming the resource elsewhere must use the same string or it will fail to correctly configure their relationships.
```
